src/test/ui/associated-type-bounds/assoc-type-eq-with-dyn-atb-fail.stderr

```diff
@@ -4,7 +4,7 @@ error[E0277]: the trait bound `String: Copy` is not satisfied
 LL |         Box::new(AssocNoCopy)
    |         ^^^^^^^^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `String`
    |
-   = note: required for the cast to the object type `dyn Bar<Assoc = <AssocNoCopy as Thing>::Out::{opaque#0}>`
+   = note: required for the cast to the object type `dyn Bar<Assoc = <AssocNoCopy as Thing>::Out::impl Copy>`
```

Comment:
This is somewhat suboptimal as you can't figure out which impl trait it is if there are multiple with the same bounds. Maybe leave out this change from this PR by adjusting path printing for opaque types?